tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py

```diff
@@ -0,0 +1,133 @@
+
+# orig_arl1s = ard_reprocessed_l1s.__wrapped__
+
+
+@pytest.fixture
+def set_up_dirs_and_db():
+    setup_script = Path(__file__).parent.joinpath("db_index.sh")
+    cmd = [setup_script]
+    try:
+        cmd_stdout = check_output(cmd, stderr=STDOUT, shell=True).decode()
+    except Exception as e:
+        print(e.output.decode())  # print out the stdout messages up to the exception
+        print(e)  # To print out the exception message
+    print("====================")
+    print(cmd_stdout)
+    print("====================")
+
+    # So the scene select call uses the correct DB
+    if "HOSTNAME" in os.environ and "gadi" in os.environ["HOSTNAME"]:
+        # Nobody call their system Brigadiers, ok.
+        end_tag = "_dev"
+    else:
+        end_tag = "_local"
+    os.environ["DATACUBE_ENVIRONMENT"] = f"{os.getenv('USER')}{end_tag}"
+    os.environ["DATACUBE_CONFIG_PATH"] = str(
+        Path(__file__).parent.joinpath("datacube.conf")
+    )
+
+
+def test_scene_move(set_up_dirs_and_db):
+    """Test the scene move function."""
+
+    current_base_path = REPROCESS_TEST_DIR
+    new_base_path = REPROCESS_TEST_DIR.joinpath("moved")
+    dry_run = False
+    product = "ga_ls9c_ard_3"
+    logdir = SCRATCH_DIR
+    scene_limit = 1
+    run_ard = False
+
+    # in bash
+    # hex=$(openssl rand -hex 3)
+    jobdir = logdir.joinpath(DIR_TEMPLATE.format(jobid=uuid.uuid4().hex[0:6]))
+    jobdir.mkdir(exist_ok=True)
+
+    cmd_params = [
+        "--current-base-path",
+        str(current_base_path.resolve()),
+        "--new-base-path",
+        str(new_base_path.resolve()),
+        "--product",
+        product,
+        "--scene-limit",
+        scene_limit,
+        "--logdir",
+        SCRATCH_DIR,
+        "--jobdir",
+        str(jobdir),
+        "--workdir",
+        SCRATCH_DIR,
+    ]
+    if run_ard:
+        cmd_params.append("--run-ard")
+    if dry_run:
+        cmd_params.append("--dry-run")
+    runner = CliRunner()
+    result = runner.invoke(ard_reprocessed_l1s, cmd_params)
+    print("***** results output ******")
+    print(result.output)
+    print("***** results exception ******")
+    print(result.exception)
+    print("***** results end ******")
+
+    # Assert a few things
+    # Two dirs have been moved
+    new_dir = REPROCESS_TEST_DIR.joinpath(
+        "moved", "ga_ls9c_ard_3", "092", "081", "2022", "06", "21"
+    )
+    fname = new_dir.joinpath(
+        "ga_ls9c_ard_3-2-1_092081_2022-06-21_final.odc-metadata.yaml"
+    )
+    assert os.path.isfile(fname) == True
+
+    new_dir = REPROCESS_TEST_DIR.joinpath(
+        "moved", "ga_ls9c_ard_3", "102", "076", "2022", "06", "27"
+    )
+    fname = new_dir.joinpath(
+        "ga_ls9c_ard_3-2-1_102076_2022-06-27_final.odc-metadata.yaml"
+    )
+    assert os.path.isfile(fname) == True
+
+    # uuids have been written to an archive file
+    filename = jobdir.joinpath(ARCHIVE_FILE)
+    temp = open(filename, "r").read().splitlines()
```

Comment:
hmmm.... prolly should use `with` when dealing with an input resource to signal the fact the contents are being used and sourced from the said input resource

Reply:
hmmmm perhaps better to explicitly specify encoding

Reply:
done, thanks